### PR TITLE
Add lammps-style Nosé-Hoover npt/nvt/nph ensemble.

### PR DESCRIPTION
**Pull request by psn417**
I've tested it on a trilinic system, and it works fine.

### CODE REVIEW

File: src/model/box.cu [hunk 1]
```diff
@@ -71,25 +71,27 @@ double Box::get_volume(void) const
 
 void Box::get_inverse(void)
 {
-  cpu_h[9] = cpu_h[4] * cpu_h[8] - cpu_h[5] * cpu_h[7];
```
Comment (brucefan1983) [inline src/model/box.cu:74]: nice fix!

File: src/main_gpumd/run.cu [hunk 2]
```diff
@@ -171,12 +171,25 @@ void Run::perform_a_run()
   clock_t time_begin = clock();
   double initial_time_step = time_step;
 
+  integrate.total_steps = number_of_steps;
```
Comment (brucefan1983) [inline src/main_gpumd/run.cu:174]: How about move this statement to the function `integrate.initialize()`?

File: src/main_gpumd/run.cu [hunk 3]
```diff
@@ -171,12 +171,25 @@ void Run::perform_a_run()
...
   for (int step = 0; step < number_of_steps; ++step) {
 
     calculate_time_step(
       max_distance_per_step, atom.velocity_per_atom, initial_time_step, time_step);
     global_time += time_step;
 
+    force.compute(
```
Comment (brucefan1983) [inline src/main_gpumd/run.cu:181]: Why add this extra force calculation here? Wouldn't it make the code 2x slower?

File: src/main_gpumd/run.cu [hunk 4]
```diff
@@ -274,7 +287,10 @@ void Run::perform_a_run()
 void Run::parse_one_keyword(std::vector<std::string>& tokens)
 {
   int num_param = tokens.size();
-  const char* param[22]; // never use more than 19 parameters
+  const int max_num_param = 32;
+  if (num_param > max_num_param)
+    PRINT_INPUT_ERROR("The number of parameters should be less than 32.\n");
+  const char* param[max_num_param]; // never use more than 19 parameters
```
Comment (brucefan1983) [inline src/main_gpumd/run.cu:293]: remove the comment here

**Comment by psn417** [comment]
I've tested it on several different cases:
1. Enthalpy figure of `iso` NPH
f1b32cf16ff821ccb8a048ad997a7db

2. Enthalpy figure of `tri` NPH
feca9e924140ddf3b97e15ea23b9d37

Volume distribution
629040f1eee795b72118f1d63f51d63

3.  None hydrostatic stress: x 500 GPa, y 500 GPa, z 600 GPa
Enthalpy
59514f4eb123afc3553acfec2ba4bed
Volume
e0a670f740cd60f68fe969c3d127806

5. xy shear stress 20 GPa
Enthalpy
7bbaac0427fb5fd40b9981bb4ff24f6
Volume
4dae9165b3cdce441ec7ae2f04f0a74

**Pull request merged**